### PR TITLE
Cargo bin project

### PR DESCRIPTION
**Pull request by carlosdagos**
This sets up the most basic project setup.

`wlroots` is commented in the dependencies so this can still compile, but there's some dependencies that will have to be dealt with somehow (Nix?) automatically to build this reliably 😄

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "wlroots-rs"]
+	path = wlroots-rs
+	url = https://github.com/swaywm/wlroots-rs
```
Comment (robbiemcmichael) [inline .gitmodules]: Need to `git add wlroots-rs` for it to be on the branch.

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "arnhem"
+description = "A modal compositor for Wayland"
+version = "0.1.0"
+authors = ["Robbie McMichael <Robbie.McMichael@cba.com.au>", "Carlos Andres D'Agostino <m@cdagostino.io>"]
```
Comment (robbiemcmichael) [inline Cargo.toml]: Probably remove my email, it's not the preferred way to contact me.

**Comment by carlosdagos** [comment]
Done @robbiemcmichael 😄

**Comment by carlosdagos** [comment]
Turns out you can specify git paths in Cargo: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories

Seems like a better alternative to submodules, which can get messy. Will fix.

**Pull request closed without merging**